index.js

```diff
@@ -93,32 +93,71 @@ const discordClient = new Client({intents});
 
 discordClient.login(DISCORD_TOKEN);
 
-discordClient.on('ready', () =>{
-    console.log('discord client ready to go!')
-    discordClient.on("messageCreate", async message => {
-       console.log('message received');
+const onMessageHandler = async message => {
+    console.log('message received');
 
-        //if the message is from a bot, exit
-        if (message.author.bot) return;
+     //if the message is from a bot, exit
+     if (message.author.bot) return;
 
-        //if the message is a direct message, exit
-        if (message.type === 'dm') return;
+     //if the message is a direct message, exit
+     if (message.type === 'dm') return;
 
-        if (message.content.startsWith('!')) {
+     if (message.content.startsWith('!')) {
+          
+             let userMessage = message.content.substring(1)
+             let userInputRegex = /^\d{5}$/
```

Comment:
I'd handle full zips just in case someone puts one in.

You can use the regex to strip off the last four if they're present.

```suggestion
             const [, zip] = /^\s*(\d{5})(-\d{4})?\s*$/.match(msg)
```